doc/manual/src/release-notes/rl-next.md

```diff
@@ -6,3 +6,16 @@
 
 - Nix now allows unprivileged/[`allowed-users`](../command-ref/conf-file.md#conf-allowed-users) to sign paths.
   Previously, only [`trusted-users`](../command-ref/conf-file.md#conf-trusted-users) users could sign paths.
+
+- Nested dynamic attributes are now merged correctly by the parser. For example:
+
+  ```nix
+  {
+    nested = { foo = 1; };
+    nested = { ${"ba" + "r"} = 2; };
+  }
+  ```
+
+  This used to silently discard `nested.bar`, but now behaves correctly.
+
+  Note that this feature is [controversial](https://github.com/NixOS/nix/pull/2089#issuecomment-382672405) and should not be relied upon too much as it makes files harder to read.
```

Comment:
````suggestion
  Note that the feature of merging multiple attribute set declarations is of questionable value, as it makes files harder to read.
  It has been around for a long time and is therefore supported for backwards compatibility, but should not be relied upon.
````

I don't think we should link to GitHub issues. Just put the information inline.

Reply:
Sorry, I edited the comment while thinking about it. Please re-check the suggestion.